### PR TITLE
Task/white238/smith fixes

### PR DESCRIPTION
**Pull request by white238**
Small fixes I noticed when updating Smith.  Nothing major.

* default to python 3 for build scripts
* allow extra cmake options in gitlab ci

### CODE REVIEW

**Review by jamiebramwell — APPROVED** [review]

File: data/input_files/default.lua [hunk 1]
```diff
@@ -4,6 +4,13 @@ dt      = 0.25
 
 output_type = "SidreVisIt"
 
+-- types = l2, full, average, min, max
+-- frequency of dump vs end value
+output_fields_json = {
+    [0] = {"temperature", "l2"},
+    [1] = {"displacement", "full"}
+}
+
```
Comment (joshessman-llnl) [inline data/input_files/default.lua]: Was this something you were testing for #465?

File: src/docs/sphinx/quickstart.rst [hunk 2]
```diff
@@ -38,6 +38,12 @@ via the `uberenv wrapper script <https://github.com/LLNL/uberenv>`_ or building
 option as building HPC libraries by hand can be a tedious process. Once the third party libraries are built, Serac can be built using the
 cmake-based `BLT HPC build system <https://github.com/LLNL/blt>`_.
 
+.. note::
+  If you get the following error ``ERROR: pip version 19.0.3 is too old to install clingo``, run the
+  following command to upgrade your pip: ``python3 -m pip install --user --upgrade pip``.  This error
+  will not necessarily be the last error on the screen.
+
+
```
Comment (joshessman-llnl) [inline src/docs/sphinx/quickstart.rst:46]: 👍

**Review by joshessman-llnl — APPROVED** [review]

File: data/input_files/default.lua (hunk 1 above)
Comment (white238) [reply data/input_files/default.lua]: oh that definitely didnt mean to be checked in... ill revert

Comment (jamiebramwell) [reply data/input_files/default.lua]: heh, whoopsies!

**Pull request merged**